controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java

```diff
@@ -407,4 +412,51 @@ public void updateStreamState(final String scopeName, final String streamName,
         }).thenApply(asyncResponse::resume)
                 .thenAccept(x -> LoggerHelpers.traceLeave(log, "updateStreamState", traceId));
     }
+
+    /**
+     * Implementation of getScalingEvents REST API.
+     *
+     * @param scopeName         The scope name of stream.
+     * @param streamName        The name of stream.
+     * @param from              DateTime from which scaling events should be displayed.
+     * @param to                DateTime until which scaling events should be displayed.
+     * @param securityContext   The security for API access.
+     * @param asyncResponse     AsyncResponse provides means for asynchronous server side response processing.
+     */
+    @Override
+    public void getScalingEvents(final String scopeName, final String streamName, final Long from, final Long to,
+                                 final SecurityContext securityContext, final AsyncResponse asyncResponse) {
+        long traceId = LoggerHelpers.traceEnter(log, "getScalingEvents");
+
```

Comment:
Add input validation for from and to params (like >=0, from < to, etc) and return 400 on error.